File: frontend/src/apply_skeleton_courses.js
import fs from 'fs';

let content = fs.readFileSync('/home/on3/DENNIS/TABLESYS/frontend/src/pages/CoursesPage.tsx', 'utf8');

// Add pageLoading state
content = content.replace(
  "const [loading, setLoading] = useState(false);",
  "const [loading, setLoading] = useState(false);\n  const [pageLoading, setPageLoading] = useState(true);"
);

// Add import for TableSkeleton
content = content.replace(
  "import { useSearchParams }",
  "import TableSkeleton from '../components/skeletons/TableSkeleton';\nimport { useSearchParams }"
);

// Update fetchCourses
content = content.replace(
  "const fetchCourses = async () => {\n      try {\n        const data = await coursesAPI.getAll() as Course[];\n        setCourses(data);\n      } catch (err) {\n        console.error('Error fetching courses:', err);\n      }\n    };",
  "const fetchCourses = async () => {\n      try {\n        const data = await coursesAPI.getAll() as Course[];\n        setCourses(data);\n      } catch (err) {\n        console.error('Error fetching courses:', err);\n      } finally {\n        setPageLoading(false);\n      }\n    };"
);

// Wrap table container in loading check
const targetString = "{/* Course Tables by Level */}";
const replaceString = "{/* Course Tables by Level */}\n          {pageLoading ? (<TableSkeleton rows={8} columns={5} />) : (\n            <Box>";

content = content.replace(targetString, replaceString);

// Close the Box check
content = content.replace(
  "            </Box>\n          )}",
  "            </Box>\n          )}\n          </Box>\n          )} // Close pageLoading Box wrapper"
);

fs.writeFileSync('/home/on3/DENNIS/TABLESYS/frontend/src/pages/CoursesPage.tsx', content);

File: frontend/src/apply_skeleton_courses.py
import re

with open('/home/on3/DENNIS/TABLESYS/frontend/src/pages/CoursesPage.tsx', 'r') as f:
    content = f.read()

content = content.replace(
    "import { useSearchParams }",
    "import TableSkeleton from '../components/skeletons/TableSkeleton';\nimport { useSearchParams }"
)

content = content.replace(
    "const [loading, setLoading] = useState(false);",
    "const [loading, setLoading] = useState(false);\n  const [pageLoading, setPageLoading] = useState(true);"
)

old_fetch = """    const fetchCourses = async () => {
      try {
        const data = await coursesAPI.getAll() as Course[];
        setCourses(data);
      } catch (err) {
        console.error('Error fetching courses:', err);
      }
    };"""

new_fetch = """    const fetchCourses = async () => {
      try {
        const data = await coursesAPI.getAll() as Course[];
        setCourses(data);
      } catch (err) {
        console.error('Error fetching courses:', err);
      } finally {
        setPageLoading(false);
      }
    };"""

content = content.replace(old_fetch, new_fetch)

# Now wrap the courses table mapping
old_tables = """          {/* Course Tables by Level */}
          <Box>"""

new_tables = """          {/* Course Tables by Level */}
          {pageLoading ? (<TableSkeleton rows={8} columns={5} />) : (
          <Box>"""

content = content.replace(old_tables, new_tables)

old_end = """              );
            })}
          </Box>"""

new_end = """              );
            })}
          </Box>
          )}"""
content = content.replace(old_end, new_end)


with open('/home/on3/DENNIS/TABLESYS/frontend/src/pages/CoursesPage.tsx', 'w') as f:
    f.write(content)

